sys/net/crosslayer/ng_netreg/ng_netreg.c

```diff
@@ -101,4 +102,30 @@ ng_netreg_entry_t *ng_netreg_getnext(ng_netreg_entry_t *entry)
     return entry;
 }
 
+int ng_netreg_calculate_csum(ng_nettype_t pseudo_hdr_type,
+                             ng_nettype_t csum_hdr_type,
+                             ng_pktsnip_t *pkt)
+{
+    switch (csum_hdr_type) {
+#ifdef MODULE_NG_ICMPV6
```

Comment:
A possible alternative to this `ifdef`approach may be to put all `csum` functions (used below) in a common place with empty bodies and the `__(weak)__` attribute to be able to overwrite them by their appropriate modules. This would remove the `ifdef`-ugliness and makes this part of code more readable. I am not sure how smart the compiler is, however, to optimize away the case statements for function calls, which have empty bodies.